spec/unit/typesafe/config/config_value_spec.rb

```diff
@@ -641,3 +647,249 @@ def n(v)
     ["a", "b", "c"].map { |x| expect(falses.get_boolean(x)).to be false }
   end
 end
+
+describe "SimpleConfigOrigin" do
+  let(:has_filename) { SimpleConfigOrigin.new_file("foo") }
+  let(:no_filename) { SimpleConfigOrigin.new_simple("bar") }
+  let(:filename_with_line) { has_filename.with_line_number(3) }
+  let(:no_filename_with_line) { no_filename.with_line_number(4) }
+
+  specify "filename matches what was specified" do
+    expect(has_filename.filename).to eq("foo")
+    expect(filename_with_line.filename).to eq("foo")
+    expect(no_filename.filename).to be nil
+    expect(no_filename_with_line.filename).to be nil
+  end
+
+  specify "description matches correctly" do
+    expect(has_filename.description).to eq("foo")
+    expect(no_filename.description).to eq("bar")
+    expect(filename_with_line.description).to eq("foo: 3")
+    expect(no_filename_with_line.description).to eq("bar: 4")
+  end
+
+  specify "origins with no line number should have line number of -1" do
+    expect(has_filename.line_number).to eq(-1)
+    expect(no_filename.line_number).to eq(-1)
+  end
+
+  specify "line_number returns the right line number" do
+    expect(filename_with_line.line_number).to eq(3)
+    expect(no_filename_with_line.line_number).to eq(4)
+  end
+end
```

Comment:
This test case is missing some stuff from upstream ...
```java
        // the filename is made absolute when converting to url
        assertTrue(hasFilename.url.toExternalForm.contains("foo"))
        assertNull(noFilename.url)
        assertEquals("file:/baz", SimpleConfigOrigin.newFile("/baz").url.toExternalForm)

        val urlOrigin = SimpleConfigOrigin.newURL(new URL("file:/foo"))
        assertEquals("/foo", urlOrigin.filename)
        assertEquals("file:/foo", urlOrigin.url.toExternalForm)
```
Is that because we don't support this functionality around URLs?  If so, it'd be worth adding a comment in this test which mentions that.